ext/case_transform/src/lib.rs

```diff
@@ -2,80 +2,124 @@
-// // camel_lower: camelCase
-use inflector::cases::camelcase::to_camel_case;
-// // camel: ClassCase (PascalCase)
-use inflector::cases::classcase::to_class_case;
-
-use ruru::{Class, Object, RString, Hash, Array, Symbol, AnyObject, VM};
-use ruru::types::ValueType;
+use inflector::cases::{camelcase, classcase, kebabcase, snakecase};
 
-class!(CaseTransform);
+use ruru::{Class, Object, VerifiedObject, RString, Hash, Array, Symbol, AnyObject};
+use ruru::types::ValueType;
+use ruru::result::Error as RuruError;
 
-methods! (
-    CaseTransform,
-    itself,
+trait Transform: Object {
+    fn transform(&self, transform_function: &Fn(String) -> String) -> AnyObject;
+}
 
-    fn deepTransformKeys(hash: Hash, block: &Fn(String) -> String) -> Hash {
-        let result = Hash::new();
+impl Transform for AnyObject {
+    fn transform(&self, _transform_function: &Fn(String) -> String) -> AnyObject {
+        self.clone()
+    }
+}
 
-        hash.unwrap().each(|key, value| {
-            let newValue = if value.ty() == ValueType::Hash { deepTransformKeys(value, block).to_any_object() } else { value };
-            let newKey = RString::new(block(key.unwrap().to_string()));
-            result.store(newKey, newValue);
-        });
+impl Transform for RString {
+    fn transform(&self, transform_function: &Fn(String) -> String) -> AnyObject {
+        let result = transform_function(self.to_string();
 
-        result
+        RString::new(&result)).to_any_object()
     }
+}
+
+impl Transform for Symbol {
+    fn transform(&self, transform_function: &Fn(String) -> String) -> AnyObject {
+        let result = transform_function(self.to_string());
 
-    fn transformArray(value: Array, transformMethod: &Fn(AnyObject) -> AnyObject) -> Array {
-        value.map(|item| transformMethod(item)).unwrap()
+        Symbol::new(&result).to_any_object()
     }
+}
+
+impl Transform for Hash {
+    fn transform(&self, transform_function: &Fn(String) -> String) -> AnyObject {
+        let mut result = Hash::new();
+
+        self.each(|key, value| {
+            let new_key = transform(key, transform_function);
+            let new_value = match value.ty() {
+                ValueType::Hash => transform(value, transform_function),
+                _ => value,
+            };
 
-    fn transformHash(value: Hash, transformMethod: &Fn(AnyObject) -> AnyObject) -> Hash {
-        deepTransformKeys(value, |key| transformMethod(key))
+            result.store(new_key, new_value);
+        });
+
+        result.to_any_object()
     }
+}
+
+impl Transform for Array {
+    fn transform(&self, transform_function: &Fn(String) -> String) -> AnyObject {
+        // Temp hack to consume &self for iterator
+        let result = unsafe { self.to_any_object().to::<Array>() };
 
-    fn transformSymbol(value: Symbol, transformMethod: &Fn(AnyObject) -> AnyObject) -> Symbol {
-        let transformed = transformMethod(value);
-        Symbol::new(transformed);
+        result.into_iter()
```

Comment:
They kinda are iterable, but after they are explicitly converted to iterators :) Same for slices, vectors etc